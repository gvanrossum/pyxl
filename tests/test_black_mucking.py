# coding: pyxl
import html
class A:
    def foo():
        lol = (<test_thing
                  test_something_whatever={{"initialsomething": initialsomething,
                                            "thisisirritating": lurrrr,
                                            "whatever": whatever,}}
              />)

        foo = (<test_thing
                  test_something_whatever={["initialsomething",
                                            "thisisirritating",
                                            "whatever",
                                            "more args",
                  ]}
              />)

        baz = (<test_thing
                  test_something_whatever={foo("initialsomething",
                                               "thisisirritating",
                                               "whatever",
                                               "more args",
                                               "again",
                                               "asdf",)}
              />)

        lol("aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa", <foo>
                                                            <bar />
                                                        </foo>)

        return (
            <foo>
                <bar baz="{baz}" spam="{spam}" eggs="{eggs}" />
            </foo>
        )
