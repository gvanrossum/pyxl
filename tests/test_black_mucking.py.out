# coding: pyxl

import html


class A:
    def foo():
        lol = (
            <test_thing
               test_something_whatever={
                                           {
                                               "aaaaaaaaaaaaaaaa": aaaaaaaaaaaaaaaa,
                                               "bbbbbbbbbbbbbbbb": lurrrr,
                                               "whatever": whatever,
                                           }
                                       }
            />
        )

        # Unfortunately black doesn't understand the real column
        # positions so does not wrap this
        foo = (
            <test_thing
               test_something_whatever={["initialsomething", "bbbbbbbbbbbbbbbb", "whatever", "more args"]}
            />
        )

        baz = (
            <test_thing
               test_something_whatever={
                                           foo(
                                               "initialsomething",
                                               "bbbbbbbbbbbbbbbb",
                                               "whatever",
                                               "more args",
                                               "again",
                                               "asdf",
                                           )
                                       }
            />
        )

        lol(
            "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
            (
                <foo>
                    <bar />
                </foo>
            ),
        )

        a = (
            <testing>
               hello {world}
            </testing>
        )

        b = (
            <testing>
                hello {
                          {
                              "initialsomething": initialsomething,
                              "bbbbbbbbbbbbbbbb": lurrrr,
                              "whatever": whatever,
                          }
                      } world
            </testing>
        )

        indented_annoyingly = (
            <frag>
                this is very popular...
            </frag>
        )

        return (
            <foo>
                <bar baz="{baz}" spam="{spam}" eggs="{eggs}" />
            </foo>
        )
